File: flink-test/src/main/java/com/atguigu/sourcetest/Source_Collection.java
package com.atguigu.sourcetest;

import com.atguigu.apitest.beans.SensorReading;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import java.util.Arrays;

/**
 * @ClassName Source_Collection
 * @Description TODO
 * @Author Xing
 * @Date 2021/4/10 23:08
 * @Version 1.0
 */
public class Source_Collection {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        DataStream<SensorReading> dss = env.fromCollection(Arrays.asList(new SensorReading("sensor1", 1547718201L, 14.0),
                new SensorReading("sensor2", 1547712401L, 12.4),
                new SensorReading("sensor3", 1547718201L, 18.8))
        );

//     TODO   以第一个元素作为 type  后面的所有元素必须和这个type相同
        DataStreamSource<Object> oss = env.fromElements(new Object(), 213, 123, 213, "12", 4, 1, 1, 1);

        dss.print("sensor");
        oss.print("oss");
        env.execute("job");

    }
}

File: flink-core/src/main/scala/com/atguigu/source/Source_Collection.scala
package com.atguigu.source
import org.apache.flink.streaming.api.scala._
/**
  * @ClassName Source_Collection
  * @Description TODO
  * @Author Xing
  * @Date 2021/4/12 10:32
  * @Version 1.0
  */
object Source_Collection {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment

    val list = List(1,2,3,4,5,6,7)

    val colSourceDS: DataStream[Int] = env.fromCollection(list)

    colSourceDS.print()

    val eleSource: DataStream[Any] = env.fromElements("Q","P",34)
    eleSource.print()

    val fileDS: DataStream[String] = env.readTextFile("flink-test/input/words.txt")

    fileDS.print("file")

    env.execute("from collection")
  }

}
